packages/react-components/theme-designer/src/components/ColorTokens/ColorTokens.tsx

```diff
@@ -1,62 +1,105 @@
+/* eslint-disable react/jsx-no-bind */
 import * as React from 'react';
 import { makeStyles } from '@griffel/react';
 import { ColorTokensList } from './ColorTokensList';
-import { Caption1 } from '@fluentui/react-components';
-import { Brands, BrandVariants, teamsLightTheme, Theme } from '@fluentui/react-theme';
+import { Button, Caption1 } from '@fluentui/react-components';
+import { Brands, BrandVariants, teamsLightTheme } from '@fluentui/react-theme';
 import { OverridableTokenBrandColors } from './OverridableTokenBrandColors';
 import { brandTeams } from '../../utils/brandColors';
-
-import type { DispatchTheme } from '../../useThemeDesignerReducer';
+import type { DispatchTheme, AppState } from '../../useThemeDesignerReducer';
+import { themeNames } from '../../utils/themeList';
 
 export interface ColorTokensProps {
   className?: string;
-  isDark: boolean;
   brand: BrandVariants;
-  dispatchState: React.Dispatch<DispatchTheme>;
+  appState: AppState;
+  dispatchAppState: React.Dispatch<DispatchTheme>;
 }
 
+export type ColorOverrideBrands = Record<string, Brands>;
+
+export type ColorOverrides = Record<string, ColorOverrideBrands>;
+
+export type DispatchColorOverrides = {
+  type: string;
+  colorToken?: string;
+  newValue?: Brands;
+};
+
+const getCurrentOverride = (appState: AppState, colorOverride: ColorOverrides) => {
+  return colorOverride[appState.themeLabel];
+};
+
+const useColorOverrideReducer = (appState: AppState, brand: BrandVariants) => {
+  const initialColorOverride: ColorOverrides = Object.fromEntries(themeNames.map(currTheme => [currTheme, {}]));
+
+  const colorOverrideReducer: (
+    state: ColorOverrides,
+    action: { type: string; colorToken?: string; newValue?: Brands },
+  ) => ColorOverrides = (state, action) => {
+    switch (action.type) {
+      case 'Add Override':
+        if (!action.colorToken || !action.newValue) {
+          return state;
+        }
+        return {
+          ...state,
+          [appState.themeLabel]: { ...state[appState.themeLabel], [action.colorToken]: action.newValue },
```

Comment:
don't pull data from global state. make the themeLabel a string passed into the action

Reply:
a reducer should have expectable results. it shouldn't rely on anything but current local state, and actions